collect_app/src/main/java/org/odk/collect/android/fragments/dialogs/CopticDatePickerDialog.java

```diff
@@ -82,6 +73,10 @@ private LocalDateTime getCurrentCopticDate() {
             copticDay = copticDate.dayOfMonth().getMaximumValue();
         }
 
+        if (copticDay < copticDate.dayOfMonth().getMinimumValue()) {
```

Comment:
Could you explain this change to me. You added a similar code to other date dialogs as well and I think I understand why we check if `copticDay > copticDate.dayOfMonth()` a few lines above since it's possible that when we change months we might move from one that has more days than another but is that possible that the number which represents months is negative?

Reply:
Hi @grzesiek2010! Some of the unit tests fails without this code. Actually when the widget does not have any answer, and the user opens up the date picker dialog, the view model sets the value of selected date as null, and we get the default value for day as 0, while it must be in the range [1,30/31] for custom date picker dialogs.

Reply:
Ok I see, thanks.